File: core/src/main/java/ir/logicfan/core/ui/base/BaseFragmentModule.java
package ir.logicfan.core.ui.base;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import com.tbruyelle.rxpermissions2.RxPermissions;
import dagger.Module;
import dagger.Provides;
import ir.logicfan.core.di.qulifier.FragmentChildFragmentManager;
import ir.logicfan.core.di.qulifier.ParentFragment;
import ir.logicfan.core.di.scope.PerFragment;

@Module
public abstract class BaseFragmentModule {

    @Provides
    @FragmentChildFragmentManager
    @PerFragment
    static FragmentManager childFragmentManager(@ParentFragment Fragment fragment) {
        return fragment.getChildFragmentManager();
    }

    @Provides
    @PerFragment
    static RxPermissions rxPermission(@ParentFragment Fragment fragment) {
        return new RxPermissions(fragment);
    }
}

File: core/src/main/java/ir/logicfan/core/ui/base/BaseFragmentModule.kt
package ir.logicfan.core.ui.base

import androidx.fragment.app.Fragment
import com.tbruyelle.rxpermissions2.RxPermissions
import dagger.Module
import dagger.Provides
import ir.logicfan.core.di.qulifier.ParentFragment
import ir.logicfan.core.di.scope.PerFragment

@Module
abstract class BaseFragmentModule {

    @Module
    companion object {

        @Provides
        @PerFragment
        @JvmStatic
        fun rxPermission(@ParentFragment fragment: Fragment): RxPermissions {
            return RxPermissions(fragment)
        }
    }
}
